.github/workflows/api-binary-compatibility.yml

```diff
@@ -38,15 +38,25 @@ jobs:
     runs-on: ubuntu-latest
     steps:
       - uses: actions/checkout@v2
+        if: github.event_name == 'pull_request'
+        with:
+          # fetch-depth of zero ensures that the tags are pulled in and we're not in a detached HEAD state
+          # revapi depends on the tags, specifically the tag from git describe, to find the relevant override
+          # in the .palantir/revapi.yml file
+          #
+          # See https://github.com/actions/checkout/issues/124
+          fetch-depth: 0
+          ref: ${{ github.event.pull_request.head.ref }}
```

Comment:
will this run on a merge commit between target branch and the PR?

Reply:
Yes, I believe it should. Somewhat hard to test admittedly. But the merge commit counts as a `push` against the target branch. For us, that’s typically master so keeping the default is fine.

I got this from the docs of the action and the associated issue. That’s why there’s a checkout if push step just below.

We can double check once it is merged in though. For our current uses (merging into master), it’ll be fine. I’ll double check whether the GitHub context has anything on push that we might grab as well though for the ref field.

Good question @findepi

Reply:
the point is -- i don't know why you want to specify `ref` at all, instead of relying on checkout action's default behavior.
if you did go with defaults, you wouldn't need these `if`s, so it would be simpler, so there must be a reason not to.
IMO the reason should be documented as a code-level comment (unless it's obvious, which doesn't seem to be the case)